### PR TITLE
{bio}[GCC/10.2.0] CIRCexplorer2 v2.3.8, Cufflinks v20190706, TopHat v2.1.2 w/ Python 2.7.18

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
requires ~~#12469~~ (segemehl), ~~#12683~~ (SciPy-bundle)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/c/CIRCexplorer2/CIRCexplorer2-2.3.8-foss-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+source_urls = [PYPI_SOURCE]
+sources = [SOURCE_TAR_GZ]
+checksums = ['8261ee884f6d895131ce681d5eda6554376e973851a9de4dc50e2b3e242ccc80']
+
+dependencies = [
+    ('Python', '3.8.6'),
+    ('SciPy-bundle', '2020.11'),
+    # sticking to v2.1.0, since using v2.1.1 has problems, cfr. https://www.biostars.org/p/186979/
+    ('TopHat', '2.1.2'),
```
Comment (boegel) [inline easybuild/easyconfigs/c/CIRCexplorer2/CIRCexplorer2-2.3.8-foss-2020b.eb:19]: Trying with TopHat v2.1.2, but may need to fall back to using v2.1.0 instead, see also https://github.com/easybuilders/easybuild-easyconfigs/pull/6554#discussion_r211962344

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node2725.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/5102483b7b4fe6ef7c20cecd858bb4a2 for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 2 out of 4 (3 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.2
See https://gist.github.com/37a28eae58ac965808788d80fede3caa for a full test report.

**Comment by verdurin** [comment]
For `TopHat`:
```
== 2021-04-19 19:55:12,218 modules.py:1347 DEBUG Output found in stdout, seems like '/usr/share/lmod/lmod/libexec/lmod python load TopHat/2.1.2-GCC-10.2.0' ran fine
== 2021-04-19 19:55:12,219 run.py:206 DEBUG run_cmd: running cmd tophat --version (in /mnt/ds918plus/eb/nuc-fedora32/software/TopHat/2.1.2-GCC-10.2.0)
== 2021-04-19 19:55:12,219 run.py:225 INFO running cmd: tophat --version
== 2021-04-19 19:55:12,319 run.py:578 DEBUG Using default regular expression: (?<![(,-]|\w)(?:error|segmentation fault|failed)(?![(,-]|\.?\w)
== 2021-04-19 19:55:12,320 easyblock.py:2846 WARNING Sanity check: sanity check command tophat --version exited with code 1 (output:   File "/mnt/ds918plus/eb/nuc-fedora32/software/TopHat/2.1.2-GCC-10.2.0/bin/tophat", line 1003
    except getopt.error, msg:
                       ^
SyntaxError: invalid syntax
)
```
which is a bit odd.

**Comment by boegel** [comment]
@verdurin Apparently TopHat requires Python 2, while your system `python` is Python 3...

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 10 out of 10 (3 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/6d6f7ec4bc77028fce2992f9af8f25a7 for a full test report.

**Comment by verdurin** [comment]
@boegel yes, clearly fine on CentOS 7 with Python 2 in the OS.

Is that why it failed on CentOS 8 too?

**Comment by boegel** [comment]
> @boegel yes, clearly fine on CentOS 7 with Python 2 in the OS.
> 
> Is that why it failed on CentOS 8 too?

Yes, well, similar: `python` is not available at all there, so the TopHat sanity check command `tophat --version` fails with:

```
/usr/bin/env: python: No such file or directory
```

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3508.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/da9e40f1a414b80b7f354228606cbfeb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3105.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/923667abd07d513d76fa85fa60f5f58d for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.2
See https://gist.github.com/d711a5eaf018e1451965debbca0e6388 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node2674.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/73ac7045cd642081ba0b0d27c6324f2c for a full test report.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 4 out of 4 (5 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/221bd93b49da0ef042117e381d41340e for a full test report.

**Review by verdurin — APPROVED** [review]
Looks fine.

**Comment by verdurin** [comment]
Going in, thanks @boegel!

File: easybuild/easyconfigs/c/CIRCexplorer2/CIRCexplorer2-2.3.8-foss-2020b.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/c/CIRCexplorer2/CIRCexplorer2-2.3.8-foss-2020b.eb]: follow up in #12857 ...

**Pull request merged**